server/sonar-server/src/main/java/org/sonar/server/custommeasure/ws/DeleteAction.java

```diff
@@ -0,0 +1,98 @@
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * SonarQube is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+package org.sonar.server.custommeasure.ws;
+
+import java.net.HttpURLConnection;
+import org.sonar.api.server.ws.Request;
+import org.sonar.api.server.ws.Response;
+import org.sonar.api.server.ws.WebService;
+import org.sonar.api.web.UserRole;
+import org.sonar.core.component.ComponentDto;
+import org.sonar.core.custommeasure.db.CustomMeasureDto;
+import org.sonar.core.permission.GlobalPermissions;
+import org.sonar.core.persistence.DbSession;
+import org.sonar.core.persistence.MyBatis;
+import org.sonar.server.db.DbClient;
+import org.sonar.server.exceptions.ServerException;
+import org.sonar.server.user.UserSession;
+
+public class DeleteAction implements CustomMeasuresWsAction {
+
+  private static final String ACTION = "delete";
+  public static final String PARAM_ID = "id";
+
+  private final DbClient dbClient;
+  private final UserSession userSession;
+
+  public DeleteAction(DbClient dbClient, UserSession userSession) {
+    this.dbClient = dbClient;
+    this.userSession = userSession;
+  }
+
+  @Override
+  public void define(WebService.NewController context) {
+    WebService.NewAction action = context.createAction(ACTION)
+      .setPost(true)
+      .setHandler(this)
+      .setSince("5.2")
+      .setDescription("Delete a custom measure.<br /> Requires 'Administer System' permission or 'Administer' permission on the project.");
+
+    action.createParam(PARAM_ID)
+      .setDescription("Id")
+      .setExampleValue("24")
+      .setRequired(true);
+  }
+
+  @Override
+  public void handle(Request request, Response response) throws Exception {
+    long id = request.mandatoryParamAsLong(PARAM_ID);
+
+    DbSession dbSession = dbClient.openSession(false);
+    try {
+      CustomMeasureDto customMeasure = searchCustomMeasure(dbSession, id);
+      checkPermissions(dbSession, customMeasure);
+      dbClient.customMeasureDao().delete(dbSession, id);
+      dbSession.commit();
+    } finally {
+      MyBatis.closeQuietly(dbSession);
+    }
+
+    response.noContent();
+  }
+
+  private CustomMeasureDto searchCustomMeasure(DbSession dbSession, long id) {
+    CustomMeasureDto customMeasure = dbClient.customMeasureDao().selectNullableById(dbSession, id);
+    if (customMeasure == null) {
+      throw new ServerException(HttpURLConnection.HTTP_CONFLICT, "Custom measure not found: " + id);
```

Comment:
I think it would be better to call ``selectById``, which correctly throws the expected ``NotFoundException``.